src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php

```diff
@@ -72,14 +73,17 @@ public function getResourceId(): string
         return (string) $this->category->getId();
     }
 
-    public function getResourceLocale(): ?string
+    public function getResourceLocale(): string
     {
         return $this->locale;
     }
 
     public function getResourceTitle(): ?string
     {
-        $translation = $this->category->findTranslationByLocale($this->locale);
+        $resourceTitleLocale = $this->getResourceTitleLocale();
+        Assert::notNull($resourceTitleLocale);
```

Comment:
Because the implementation of the `getResourceTitleLocale(): ?string` method is in the `DomainEvent` class now, we can't typehint it with `getResourceTitleLocale(): string` anymore, therefore we need to ensure, that the `resourceTitleLocale` is not null

Reply:
i think this would be a second reason for using `$this->locale` 🙂

Reply:
we can use a test to ensure that the locale used by the `getResourceTitle()` matches the `DomainEvent::getResourceTitleLocale()`

Reply:
I really think we should use the `getResourceTitleLocale()` method here:
1. Errors will occurr earlier and won't lead to inconsitent data in the database
2. I think the name of the `getResourceTitleLocale()` method is quite self explanatory, so IMO, here it's easier to understand what's actually going on

Reply:
I would also not call the public `getResourceLocale` internal and directly accessing `$this->locale` or the correct locale here.

Reply:
> 1. Errors will occurr earlier and won't lead to inconsitent data in the database, if someone forgot to implement the `getResourceTitleLocale()` method

If somebody forgets to implement the `getResourceTitleLocale()` method, it means that he is implementing his own domain event. this means he will not use this class at all, right? 🤔 

> 2. I think the name of the `getResourceTitleLocale()` method is quite self explanatory, so IMO, here it's easier to understand what's actually going on

Might be a matter of taste. I feel like its harder to understand because I need to have a look at the base class to check what the `getResourceTitleLocale()` is doing. Also I will probably wonder about why the assertion is needed at first 🙈

Reply:
> If somebody forgets to implement the getResourceTitleLocale() method, it means that he is implementing his own domain event. this means he will not use this class and its implementation at all, right? 🤔

But he will probably use this or a similar class as reference

Reply:
looks like you did forget to change it here to $this->locale?

Reply:
oops 🙈 fixed now